### DISCUSSION TITLE
How to pair secondary device?

### DISCUSSION THREAD

**Discussion by shalak** [bbernhard/signal-cli-rest-api#618] [Q&A] [answered]
I have the REST API running as a home assistant addon, I did the registration using a fresh phone number there. I'd like to install the Signal Desktop and hook it up with this account. Is it possible?

**Comment by shalak** [comment, marked as answer]
Found the answer [in another chain](https://github.com/haberda/signal-addon/issues/12).

Steps I took:

- initiated the pairing process in the desktop app
- saved the QR code to JPEG file
- used simple python script to scan it:
```python
import cv2 as cv
from pyzbar.pyzbar import decode
import argparse

# Set up argument parser
parser = argparse.ArgumentParser(description="Decode QR Code from an image file.")
parser.add_argument("path", type=str, help="Path to the QR code image file.")
args = parser.parse_args()

# Read the image
qrcode_img = args.path
img = cv.imread(qrcode_img)

if img is None:
    print(f"Error: Could not read image from path '{qrcode_img}'")
    exit(1)

# Decode the QR code
decoded_data = decode(img)

if not decoded_data:
    print("No QR code detected in the image.")
    exit(1)

# Parse and print the decoded data
url = decoded_data[0].data.decode()
print(f"URL:\n{url}")
```
- copied the URL
- since I'm running CLI REST as a Hassio Addon on HassOD, I SSHed into the "real" system and `docker exec -it <something>_signal_addon bash`:
```
su signal-api
signal-cli --config /config -u "+ACCOUNT_NUMBER" addDevice --uri "URL_GOES_HERE"
```
- the above showed me the following log:
```
INFO  SignalAccount - Config file is in use by another instance, waiting…
```
- run `ps aux` and found the PID of signal java process
- `kill <PID>`, the desktop app started pairing process, appeared to be stuck in contacts sync stage
- run `signal-cli --config /config -u "+ACCOUNT_NUMBER" sendContacts`
- desktop app finished pairing
- restarted the docker add-on from Hassio GUI

**Comment by shalak** [comment]
I tried this again, this time instead of the Desktop version, it was an iPad app. Unfortunately I kept getting:

```
ERROR AddDeviceCommand - Add device link failed: [404] Not found
Add device link failed
```

It might be related to the fact that the QR code on the iPad changes very fast? Any ideas on how to solve this issue?

**Reply by shalak** [reply]
I'm doing it within 5s on a fresh QR code, that cannot be the case...

**Reply by bbernhard** [reply]
Does this also happen with this version: `bbernhard/signal-cli-rest-api:1747169206-ci`?

**Reply by shalak** [reply]
Hmm, is there a way for me to bump the CLI in the hassio-addon container?
